### PR TITLE
nfc: ndef: align file and API naming

### PR DESCRIPTION
**Pull request by kapi-no**
This change aligns file and API naming for NFC NDEF category.

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

### CODE REVIEW

**Comment by kapi-no** [comment]
CC: @grochu, @KAGA164

**Comment by KAGA164** [comment]
I am wondering if we need `ndef ` prefix in records API, maybe it still be understandable without it ?

**Comment by kapi-no** [comment]
> 
> 
> I am wondering if we need `ndef ` prefix in records API, maybe it still is understandable without it?

That could be part topic of another PR. The main purpose of this one is to make all modules consistent (using the same style). At the moment, some records were using prefix and others were not.

You probably suggested to remove ndef prefix to make API names shorter. I`m curious what others think of it. It's always a tradeoff:
consistency / clarity (with ndef, longer names)
vs
ease of use (without ndef, shorter names)

File: include/nfc/ndef/text_rec.h [hunk 1]
```diff
@@ -64,32 +64,32 @@ struct nfc_text_rec_payload_desc {
  * @retval 0 If the operation was successful.
  *           Otherwise, a (negative) error code is returned.
  */
-int nfc_text_rec_payload_constructor(
-		struct nfc_text_rec_payload_desc *nfc_rec_text_payload_desc,
+int nfc_ndeF_text_rec_payload_encode(
```
Comment (joerchan) [inline include/nfc/ndef/text_rec.h:67]: Fix the capital `F`

File: subsys/nfc/ndef/text_rec.c [hunk 2]
```diff
@@ -15,19 +15,19 @@
...
+	    struct nfc_ndef_text_rec_payload const *nfc_rec_text_payload_desc)
 {
 	return (TEXT_REC_STATUS_SIZE +
 		nfc_rec_text_payload_desc->lang_code_len +
 		nfc_rec_text_payload_desc->data_len);
 }
 
-int nfc_text_rec_payload_constructor(
-		struct nfc_text_rec_payload_desc *nfc_rec_text_payload_desc,
+int nfc_ndeF_text_rec_payload_encode(
```
Comment (joerchan) [inline subsys/nfc/ndef/text_rec.c:29]: Fix capital `F`

**Comment by joerchan** [comment]
@grochu @KAGA164 We will need a changelog entry for this PR I think.
I suggest making the decision about whether `ndef` should be part of the name in this PR in that case to avoid going over it twice.

**Comment by kapi-no** [comment]
> 
> 
> @grochu @KAGA164 We will need a changelog entry for this PR.
> I suggest making the decision about whether `ndef` should be part of the name in this PR in that case to avoid going over it twice.

I, @KAGA164 and @grochu discussed it in a call and we decided to keep NDEF API names as they were suggested in this PR.

File: include/nfc/ndef/text_rec.h (hunk 1 above)
Comment (kapi-no) [reply include/nfc/ndef/text_rec.h:67]: thanks, I will fix it soon

Comment (kapi-no) [reply include/nfc/ndef/text_rec.h]: fixed

**Pull request merged**